### PR TITLE
feat: Initial layout of CEEI dashboard

### PR DESCRIPTION
**Pull request by lrt-sibat**
Thank you for submitting a pull request!

## Description
Creating this PR to coordinate with @mikedstr for the CEEI Dashboard.

This PR will be marked as a draft until it is ready and the Electricity Dashboard is merged.

### Type of Pull Request

* [x] Adding a feature
* [ ] Fixing a bug
* [ ] Maintaining documents
* [ ] Adding tests
* [ ] Others ()

### Verify the followings

* [ ] Code is up-to-date with the `develop` branch
* [ ] No build errors after `pnpm build`
* [ ] No lint errors after `pnpm lint`
* [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`
* [ ] Make sure all the existing features working well

### Changesets

* [ ] If your PR makes a change under `packages` folder that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

Refer to [CONTRIBUTING.MD](https://github.com/UNDP-Data/geohub/blob/develop/CONTRIBUTING.md) for more information.

┆Issue is synchronized with this [Wrike task](https://www.wrike.com/open.htm?id=1375526519) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by lrt-sibat** [comment]
Hi @JinIgarashi, I created a new PR that branches off of `develop` now instead of `develop-ibm`. Sorry for the mix-up!

File: .changeset/good-eagles-play.md
Comment (JinIgarashi) [inline .changeset/good-eagles-play.md:5]: Could you merge two changesets file into a changeset file?

also change major to minor. This PR cannot be a major release of geohub. it is just a new feature.

change changelog text like `feat: added new CEEI dashboard`

**Review by JinIgarashi — CHANGES_REQUESTED** [review]
@lrt-sibat @mikedstr 

Is source code the latest?

There are several things need to be changed before merging to production.

- a landing page is needed for CEEI dashboard

No users know what this abbreviation of CEEI is. We need a landing page like Electricity dashboard to explain what this dashboard is, what CEEI is.

- about `sites/geohub/static/geohub.json` file

we don't want to add this 70MB file in geohub repository. this kind of file should be stored in Azure Blob Storage. Could you delete these two geohub.json files?

- https://github.com/UNDP-Data/geohub/blob/feat/ceei-dashboard/sites/geohub/src/routes/(map)/dashboards/ceei/geohub.json
- https://github.com/UNDP-Data/geohub/blob/feat/ceei-dashboard/sites/geohub/static/geohub.json

then, please use this file hosted in Azure blob storage (I just uploaded current json file to Azure).

https://undpgeohub.blob.core.windows.net/ceei/ceei-data.geojson

Some buttons are not working.

- `Information` menu do nothing
image

- 2nd button from the right (property setting button) has no any action
image

- `.exc` extension mentioned in the text should be changed to `.csv`

image

**Comment by JinIgarashi** [comment]
I deployed this feature branch as a deployment slot in Azure App Service. It can be accessed from the below URL.

https://geohub-prod-ceei.azurewebsites.net/dashboards/ceei

**Comment by lrt-sibat** [comment]
Hi Jin, I'm currently merging Mike's changes with my own atm, I'll be sure to let you know when it's pushed.

File: packages/svelte-undp-design/src/lib/Carousel/Carousel.svelte
Comment (JinIgarashi) [inline packages/svelte-undp-design/src/lib/Carousel/Carousel.svelte:11]: @lrt-sibat could you update `FluidCarousel.svelte` instead of adding this one?

https://github.com/UNDP-Data/geohub/blob/feat/ceei-dashboard/packages/svelte-undp-design/src/lib/FluidCarousel/FluidCarousel.svelte

your new `Carousel` component looks almost same with FluidCarousel.

File: packages/svelte-undp-design/src/lib/Button/Button.svelte
Comment (JinIgarashi) [inline packages/svelte-undp-design/src/lib/Button/Button.svelte:12]: @lrt-sibat please also add a changeset file for `svelte-undp-design` package. otherwise, it will not be released to NPM

Comment (lrt-sibat) [reply packages/svelte-undp-design/src/lib/Button/Button.svelte:12]: Hi Jin, a changeset file was already created [here](https://github.com/UNDP-Data/geohub/blob/feat/ceei-dashboard/.changeset/eleven-houses-occur.md)

File: packages/svelte-undp-design/src/lib/Carousel/Carousel.svelte
Comment (JinIgarashi) [reply packages/svelte-undp-design/src/lib/Carousel/Carousel.svelte:11]: @lrt-sibat I think your new Crousel.svelte is not used anywhere, right? please remove it if it is not used.

File: packages/svelte-undp-design/src/lib/Button/Button.svelte
Comment (JinIgarashi) [reply packages/svelte-undp-design/src/lib/Button/Button.svelte:12]: @lrt-sibat ok. I updated your changeset files

**Review by JinIgarashi — CHANGES_REQUESTED** [review]
@lrt-sibat please fix the following points

## fixes for attribute popup

Showing attributes by clicking looks great. Thanks for that. Could you fix the following points for popup?

-	Add x button on the top-right of popup to close it.
-	Set max-height to popup content to enable scrolling vertically
-	Is it possible to change border of polygon which user clicked.

3rd feature of changing border color can be implemented by using feature state of maplibre.
See the below example. This is to change border color when it is hovered. But feature state is also available for ‘click’.
https://maplibre.org/maplibre-gl-js/docs/examples/hover-styles/

## fixes for layer name

is it possible to edit the layer name after duplicating? 

## geohub.json file

Please delete this file from repository as I mentioned before.

## add CEEI to footer

add this page link to footer

https://github.com/UNDP-Data/geohub/blob/5b1613832ea39f89b57f6ed48f457389d71a9bc7/sites/geohub/src/lib/server/config/FooterItems.ts#L45-L56

## Colormap picker

Also, it would be nice if we can change colormap for layer. This can be useful when users duplicate a layer to differentiate them.
 
ColormapPicker is available at svelte-undp-components (https://svelte-components.undpgeohub.org/?path=/docs/components-colormappicker--docs)

**Comment by JinIgarashi** [comment]
@lrt-sibat I found a bug of attribute popup.

steps to reproduce it:

1. duplicate a layer
2. customise duplicated layer to simulate
3. click anywhere on the map to show attribute popup
4. compare two layers' values, and discovered duplicated layers' values are not changed after simulating.

image

File: sites/geohub/package.json
Comment (JinIgarashi) [inline sites/geohub/package.json:139]: @lrt-sibat could you remove `@maplibre/maplibre-gl-compare` from dependencies for now?

**Comment by JinIgarashi** [comment]
@lrt-sibat also, color map is not duplicated to new layer.

- change colormap in base layer
- duplicate base layer,
- make base layer invisible
- you can realize color of duplicated layer on map is not copied

image

also, please solve confict of `pnpm-lock.yaml` by following command

```
rm pnpm-lock.yaml
pnpm i
```

File: sites/geohub/src/routes/(map)/dashboards/ceei/+page.svelte
Comment (JinIgarashi) [inline sites/geohub/src/routes/(map)/dashboards/ceei/+page.svelte:158]: @lrt-sibat can you set maxBounds to maplibre Map object to prevent users to move to outside of Africa?

https://maplibre.org/maplibre-gl-js/docs/API/type-aliases/MapOptions/#maxbounds

File: sites/geohub/src/routes/(map)/dashboards/ceei/+page.svelte
Comment (JinIgarashi) [inline sites/geohub/src/routes/(map)/dashboards/ceei/+page.svelte:177]: close button is required for popup

**Comment by JinIgarashi** [comment]
@lrt-sibat I think you can remove `All` from the dropdown menu.

- show all countries if a country is not selected
- show only selected country if a country is selected
- show all countries after clicking `x` button to reset

furthermore, I think you can remove `select` box element. Instead showing text box to filter always. If user types any keyword, show dropdown menu as it does currently.

image

**Review by JinIgarashi — APPROVED** [review]
@lrt-sibat Thanks, it looks very good now.

**Comment by JinIgarashi** [comment]
@lrt-sibat there are two small requests

## Rename `Base layer` to CEEI

- current
image

- proposed
image

## make height of popup smaller

The height of popup is too large, I think we need to set max-height to make popup smaller, and set overflow-y to enable scrolling. this might be better.

Also, maybe accordion can be introduced for `Potential`, `Urgent` and `Means and Resources` categories level.

- current

image

File: sites/geohub/package.json
Comment (JinIgarashi) [inline sites/geohub/package.json:143]: @lrt-sibat Please move `@turf/turf` to `devDependencies`. libraries used in frontend should not be included in `dependencies`.

Also, please consider using [@turf/bbox](https://www.npmjs.com/package/@turf/bbox) other adding the heavy turf.js package if you only need `bbox` feature.

**Review by JinIgarashi — CHANGES_REQUESTED** [review]
@lrt-sibat Please fix turf.js's dependencies issue. otherwise we cannot merge this.

File: sites/geohub/src/routes/(map)/dashboards/ceei/components/PopupTable.svelte
Comment (JinIgarashi) [inline sites/geohub/src/routes/(map)/dashboards/ceei/components/PopupTable.svelte:85]: @lrt-sibat Is it possible to expand only a pillar group? for example, as default, it expands `Potential`, but `Urgent` and `Means and Resources` are collapsed. If `Urgent` is expanded, other two will be collapsed.

**Review by JinIgarashi — APPROVED** [review]
just a small request for popup table behaviour, but it looks very good now.

**Pull request merged**
